src/markup/commands/helpers.ts

```diff
@@ -209,12 +209,86 @@ export const wrapPerLine =
 
             const isSingleEmptyLine =
                 range.anchor === range.head && state.doc.lineAt(range.head).length === 0;
+
             if (isSingleEmptyLine) {
-                changes.push({from: range.head, insert: before});
+                // Get markers from previous line for empty line case
+                if (range.head > 1) {
+                    const prevLine = state.doc.line(range.head - 1);
+                    const markers = parseMarkers(prevLine.text);
+                    const indentation = markers.join('');
+                    changes.push({from: range.head + indentation.length, insert: before});
+                } else {
+                    changes.push({from: range.head, insert: before});
+                }
             } else {
+                // Calculate indentation from the first line
+                const firstLine = state.doc.lineAt(range.from);
+                let indentation = '';
```

Comment:
Let`s move code to `getIndentation ` or `calculacteIndentation ` or `parseIndentation`
and tests plz